lib/galaxy/dataset_collections/structure.py

```diff
@@ -177,19 +177,14 @@ def tool_output_to_structure(get_sliced_input_collection_type, tool_output, coll
     if not tool_output.collection:
         tree = leaf
     else:
-        collection_type_descriptions = collections_manager.collection_type_descriptions
         # Okay this is ToolCollectionOutputStructure not a Structure - different
         # concepts of structure.
-        if tool_output.dynamic_structure:
-            # Two cases collection_type_source and collection_type right?
-            tree = UnitializedTree(collection_type_descriptions.for_type_description("list"))  # list is obviously wrong...
+        structured_like = tool_output.structure.structured_like
```

Comment:
https://github.com/galaxyproject/galaxy/pull/5431 made me relook at at this. So the problem here is that we should be filling out the backbone of the collection if it has a structured_like tag. This allows us to create more the jobs upfront and eases the workflow scheduling burden. If a workflow has no dynamic structures - Galaxy can prebuild all the collections and jobs - and just rely on job polling instead of workflow and job polling. Users can see more of their collections ahead of time as well.

I can take a stab at this and see if I can rearrange things. Certainly what I had before was broken - I knew it in that comment up there.

Reply:
I've got it all in my head for just one second... I'm going to loose it so jotting it down here:

- In execute sliced_input_collection_type should be sliced_input_collection_structure. That structure should never be uninitialized.
- Pass that function in to here - use it for structured_like if we have it. If not we still need to check for collection_type_source and grab the structure but return an un-initialized tree with the same structure.

That would make structured_like work again and make type_source I think.

I'll work on this - but let me know if it sounds crazy wrong.

Reply:
https://github.com/galaxyproject/galaxy/commit/76ca596807e5974de8c64b29b56d7b5b1fcf1c23